Dev backup

happy-land opened the pull request.

happy-land committed:
Branch develop

happy-land committed:
header

happy-land committed:
remove unused files

happy-land committed:
login page setup

happy-land committed:
Merge pull request #1 from happy-land/feature/login

login page setup

trashmarket committed:
button id done with collect jwt token

happy-land committed:
Merge branch 'develop' into nikita-button-id

happy-land committed:
Merge pull request #3 from happy-land/nikita-button-id

button id done with collect jwt token

happy-land committed:
scss styles

happy-land committed:
Merge pull request #4 from happy-land/ruslan-css

scss styles

Dumisel committed:
верстка footer/header

happy-land committed:
Merge pull request #5 from happy-land/lisa-css-footer/header

верстка footer/header

Dumisel committed:
DetailPage(хардкор данные, переключение тем)

happy-land committed:
profile cards, onMouseOver done

annasilina committed:
fixed ui for button.tsx

annasilina committed:
fixed ui for button.tsx

happy-land committed:
Merge pull request #7 from happy-land/lisa-detail-css

DetailPage(хардкор данные, переключение тем)

happy-land committed:
Merge pull request #8 from happy-land/anna-button-ui

Anna button UI

happy-land committed:
profile cards - desktopMode added

MaximTutunov committed:
Admin Students Component

happy-land committed:
Merge pull request #9 from happy-land/AdminStudentsList-Page

Admin students list page

happy-land committed:
comment-block draft

happy-land committed:
conflicts with delelop branch fixed

happy-land committed:
package.json fixed

happy-land committed:
Merge pull request #10 from happy-land/ruslan-main-page

Ruslan main page

happy-land committed:
utils/api.ts draft

happy-land committed:
Merge pull request #11 from happy-land/ruslan-api

utils/api.ts draft

happy-land committed:
app-routes.tsx HomePage added

happy-land committed:
Merge pull request #12 from happy-land/app-routes-hotfix

app-routes.tsx HomePage added

MaximTutunov committed:
Административная страница модерирование коммент

happy-land committed:
Merge pull request #13 from happy-land/admin-comments-list

Admin comments list

happy-land committed:
link style fixed

annasilina committed:
first steps at profile page form CSS

trashmarket committed:
select 0.0.1 done

happy-land committed:
emoji styling in progress

happy-land committed:
added scrollbar in comments list

MaximTutunov committed:
admin students comments fix

Dumisel committed:
изменения на странице detail, header, счетчик реакций

trashmarket committed:
calendare #0.0.1

happy-land committed:
emoji clicking logic no API

happy-land committed:
Merge pull request #16 from happy-land/ruslan-main-page

Ruslan main page

happy-land committed:
Merge pull request #15 from happy-land/lisa-details

изменения на странице detail, header, счетчик реакций

happy-land committed:
reactions with api draft

happy-land committed:
merge develop into ruslan-main-page

happy-land committed:
Merge branch 'develop' into admin-students-comments-fix

happy-land committed:
Merge pull request #14 from happy-land/admin-students-comments-fix

Admin students comments fix

happy-land committed:
merge from admin-students-comments-fix

happy-land committed:
comments-block in progress

happy-land committed:
data for comments and emoji from api

happy-land committed:
Merge pull request #17 from happy-land/ruslan-main-page

Ruslan main page

happy-land committed:
fix styles profile-list profile-card

annasilina committed:
added map page

happy-land committed:
test buttons removed

happy-land committed:
Merge pull request #18 from happy-land/ruslan-main-page

Ruslan main page

annasilina committed:
added apiKey for map element

Dumisel committed:
логика с реакциями

annasilina committed:
fixed apikey for map element

happy-land committed:
Merge branch 'develop' into lisa-emoji

happy-land committed:
Merge pull request #19 from happy-land/lisa-emoji

логика с реакциями

happy-land committed:
comments-block hotfix

happy-land committed:
Merge pull request #21 from happy-land/comments-block-hotfix

comments-block hotfix

happy-land committed:
Merge branch 'develop' into anna-map-page

happy-land committed:
Merge pull request #20 from happy-land/anna-map-page

Anna map page

annasilina committed:
added directory for Api class
fixed common types

MaximTutunov committed:
fix1 enter the commit message for your changes. Lines starting

MaximTutunov committed:
image restored

MaximTutunov committed:
fixed

annasilina committed:
added all api requests

annasilina committed:
added type for getCohortData return

trashmarket committed:
fix colendar whith close

annasilina committed:
added types for all Api methods
added setToken method - for token visibility inside Api
added types for all objects

annasilina committed:
deleted old api.ts

annasilina committed:
fixed map page: deleted map component, added useState for data at MapPage

annasilina committed:
Merge remote-tracking branch 'origin/anna-map-page' into anna-map-page

Dumisel committed:
DetailPage счетчик, черновик routes

trashmarket committed:
form #0.0.1

happy-land committed:
Merge pull request #23 from happy-land/admin-page

Admin page

happy-land committed:
conflicts resolved

happy-land committed:
Merge pull request #24 from happy-land/anna-api

Anna api

happy-land committed:
conflicts resolved

happy-land committed:
Merge pull request #25 from happy-land/lisa-detailpage-more

DetailPage счетчик, черновик routes

happy-land committed:
conflicts resolved

happy-land committed:
Merge pull request #26 from happy-land/nikita-form-page

Nikita form page

happy-land committed:
Merge pull request #27 from happy-land/anna-map-page

Anna map page

annasilina committed:
fixed constants - deleted extra types
fixed types.ts
added path for Link to Map Page at Home page
added htmlType for Button
deleted "react-yandex-maps" package

annasilina committed:
fixed Api methods
fixed baseUrl in api config
fixed proxy url in package json

annasilina committed:
deleted constants.tsx
fixed styles for map's avatars

happy-land committed:
Merge pull request #28 from happy-land/anna-fix

fixed constants - deleted extra types

Dumisel committed:
исправлены апи

happy-land committed:
main page fix draft

happy-land committed:
Merge branch 'develop' into lisa-review

happy-land committed:
Merge pull request #29 from happy-land/lisa-review

исправлены апи

Dumisel committed:
рутс + обновленные тестовые данные

Dumisel committed:
Merge branch 'lisa-review' of https://github.com/happy-land/visitki into lisa-review

Dumisel committed:
типы

happy-land committed:
in progress

happy-land committed:
Merge pull request #30 from happy-land/lisa-review

Lisa review

happy-land committed:
fixing

happy-land committed:
error

happy-land committed:
main fixed

happy-land committed:
Merge pull request #31 from happy-land/ruslan-main-fix

Ruslan main fix

happy-land committed:
Merge branch 'main' into develop

annasilina committed:
fixed css emoji module
added project info to README

annasilina committed:
added info about current functionality to README
fix config for Api
fixed types
fixed Map Page
changed request data method at admin components
fixed unused imports in components
fixed duplicate declaration in type TReaction

happy-land committed:
Merge pull request #32 from happy-land/develop

Develop

annasilina committed:
added some info to readme

Dumisel committed:
anna-changes

Dumisel committed:
Merge pull request #41 from happy-land/dev-new

anna-changes

gennady-bars commented on src/utils/api.ts:
```diff
@@ -0,0 +1,51 @@
+
+
+export const getStudentById = (id: string) => fetch(`/profiles/${id}`)
+.then(checkResponse)
```
**Можно лучше**

Можно сделать **универсальную функцию запроса с проверкой ответа**, чтобы не дублировать эту проверку в каждом запросе:

```js

  function request(url, options) {
    // принимает два аргумента: урл и объект опций, как и `fetch`
    return fetch(url, options).then(checkResponse)
  }

```

И теперь просто нужно заменить все `fetch` на `request` и убрать дублирование проверки на `ok`. Все остальное будет без изменений. Код станет чище

gennady-bars commented on src/api/Api.ts:
```diff
@@ -0,0 +1,154 @@
+				...this.headers,
+				Authorization: `Bearer ${this.token}`
+			}
+		}).then((response) => this.checkResponse(response));
```
Вам нужно как-то определиться. Либо это класс `Api` занимается запросами, либо файл `api.ts` с функциями запросов.

gennady-bars commented on src/api/Api.ts:
```diff
@@ -0,0 +1,154 @@
+				...this.headers,
+				Authorization: `Bearer ${this.token}`
+			}
+		}).then((response) => this.checkResponse(response));
```
**Можно лучше** 

Можно сделать **универсальный метод запроса с проверкой ответа**, чтобы не дублировать эту проверку в каждом запросе:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

И теперь просто нужно заменить все `fetch` на `this._request`. Все остальное будет без изменений

gennady-bars commented on src/components/admin-comments-list/admin-comments-list.tsx:
```diff
@@ -0,0 +1,64 @@
+    //   .finally(() => {});
+    api.getCommentsData().then((data) => {
+      setComments([...data.items])
+    })
```
Для любых запросов на сервер в **самом конце цепочки** запроса нужно осуществлять проверку на возможные ошибки с помощью `catch`.

**Нужно исправить это везде по коду в конце запросов.**

gennady-bars commented on src/components/app/app.tsx:
```diff
@@ -0,0 +1,15 @@
+
+const App: FC = () => {
+  return (
+    <div>
```
пустые `<div>` без стилей по семантике лучше не делать. Они не нужны

gennady-bars commented on src/components/comments-block/commets-block.tsx:
```diff
@@ -0,0 +1,68 @@
+    if(user.tag === 'admin' || owner === true) {
+    api.getReactionsForUser(user._id)
+      .then((data) => {
+        setComments(data.items.filter((el: { text: string; }) => el.text));
```
Вам больше **не нужно типизировать** аргументы в `map`, `filter`, `reduce` и тд, если массив уже 1 раз типизирован. Тайпскрипт возьмет тип автоматически. Код станет чище


```js

        setComments(data.items.filter((el) => el.text));


```

gennady-bars commented on src/components/comments-block/commets-block.tsx:
```diff
@@ -0,0 +1,68 @@
+      .catch(err => console.log(err));
+  }}, [isOpen])
+
+  const handleSubmit = (event: any) => {
```
`(event: any)`


Все события обработчиков нужно **правильно** типизировать. Это легко сделать. Для `onChange` есть тип `ChangeEvent<HTMLInputElement>`, а для `onSubmit` есть `FormEvent<HTMLFormElement>` в библиотеке `react`. Для обычных функций есть тип `SyntheticEvent` для их событий. Тип `Response` для ответа от сервера, тип `KeyboardEvent` для событий клавиатуры, тип `RouteProps` для рутов

Есть небольшая хитрость: можно всегда проверить, какой тип у функции или события. Для этого нужно навести мышкой на атрибут `onChange`, `onSubmit`, `onClick` и тд. Там высветится типизация того, что принимают эти атрибуты.

gennady-bars commented on src/components/comments-block/commets-block.tsx:
```diff
@@ -0,0 +1,68 @@
+              <div className={styles.line}></div>
+            </li>)}
+          </ul>
+          <form className={styles.form} onSubmit={event => handleSubmit(event)}>
```
`onSubmit={event => handleSubmit(event)`

можно упростить запись


```js

 onSubmit={handleSubmit}

```

gennady-bars commented on src/components/login-form/login-form.tsx:
```diff
@@ -0,0 +1,38 @@
+  const login = useCallback<TLoginCallback>((event) => {
+    event.preventDefault();
+    console.log('PRIVET');
+    fetch('http://localhost:3000/users')
```
`    fetch('http://localhost:3000/users')
`

это вряд ли будет работать 

наверное, нужно другой урл указать

gennady-bars commented on src/components/login-form/login-form.tsx:
```diff
@@ -0,0 +1,38 @@
+  return (
+    <div>
+      <h1 className={styles.title}>С кем я учусь?</h1>
+      <Button handleClick={login} htmlType='submit'>
```
`htmlType='submit'`


на кнопку сабмита не вешают клики.


и тут нет тега `form`, значит, тут кнопка должна иметь тип `button`

а чтобы перезагрузилась страница, можно как-то по-другому это обыграть


хотя в реакте не должно быть перезагрузки страницы.

gennady-bars commented on src/components/profile-list/profile-list.tsx:
```diff
@@ -0,0 +1,177 @@
+    window.addEventListener('resize', (event) => handleWindowResize(event));
+
+    return () => {
+      window.removeEventListener('resize', (event) => handleWindowResize(event));
```
`      window.removeEventListener('resize', (event) => handleWindowResize(event));
`

в таком варианте невозможно удалить обработчик, так как Вы его **ссылку** не передаете. Там Вы передаете **анонимную** стрелочную функцию (у неё нет постоянной ссылки). Движок браузера не сможет найти по ссылке ее и удалить. Ошибки никакой браузер не выдаст, так как ему без разницы, что он не нашел функцию.



Подробнее про правильное удаление обработчиков тут: https://learn.javascript.ru/introduction-browser-events#addeventlistener

gennady-bars commented on src/components/profile-list/profile-list.tsx:
```diff
@@ -0,0 +1,177 @@
+    document.addEventListener('scroll', (event) => scrollHandler(event));
+
+    return () => {
+      document.removeEventListener('scroll', (event) => scrollHandler(event));
```
в таком варианте невозможно удалить обработчик, так как Вы его **ссылку** не передаете. Там Вы передаете **анонимную** стрелочную функцию (у неё нет постоянной ссылки). Движок браузера не сможет найти по ссылке ее и удалить. Ошибки никакой браузер не выдаст, так как ему без разницы, что он не нашел функцию.

gennady-bars commented on src/components/student-details/student-details.tsx:
```diff
@@ -0,0 +1,256 @@
+    window.addEventListener('resize', (event) => handleWindowResize(event));
+
+    return () => {
+      window.removeEventListener('resize', (event) => handleWindowResize(event));
```
в таком варианте невозможно удалить обработчик, так как Вы его **ссылку** не передаете. Там Вы передаете **анонимную** стрелочную функцию (у неё нет постоянной ссылки). Движок браузера не сможет найти по ссылке ее и удалить. Ошибки никакой браузер не выдаст, так как ему без разницы, что он не нашел функцию.

gennady-bars commented on src/ui/widget-date/input-date.tsx:
```diff
@@ -0,0 +1,238 @@
+
+    document.addEventListener('click', clickPad)
+
+    return () => document.removeEventListener('click', clickPad)
```
Вот тут правильно удаляется обработчик

gennady-bars commented on src/components/student-details/student-details.tsx:
```diff
@@ -0,0 +1,256 @@
+	useEffect(() => {
+    if (desktopMode) {
+      setChatIconStylePhoto({ display: 'none' }); 
+			setChatIconStyleQuote({ display: 'none' });
+			setChatIconStyleHobby({ display: 'none' });
+			setChatIconStyleStatus({ display: 'none' });
+			setChatIconStyleJob({ display: 'none' });
+			setChatIconStyleEdu({ display: 'none' });
```
**Можно лучше**

тут можно было один показатель создать для отображения  и использовать его везде. Не стоит дублировать одинаковые данные в каждом отдельном случае

gennady-bars commented on src/components/student-details/student-details.tsx:
```diff
@@ -0,0 +1,256 @@
+			setChatIconStyleJob({ display: 'none' });
+			setChatIconStyleEdu({ display: 'none' });
+    } else {
+			setChatIconStylePhoto({ display: 'flex' }); 
+			setChatIconStyleQuote({ display: 'flex' });
+			setChatIconStyleHobby({ display: 'flex' });
+			setChatIconStyleStatus({ display: 'flex' });
+			setChatIconStyleJob({ display: 'flex' });
+			setChatIconStyleEdu({ display: 'flex' });
```
**Можно лучше**

тут можно было один показатель создать для отображения  и использовать его везде. Не стоит дублировать одинаковые данные в каждом отдельном случае

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная, но есть некоторые недочеты: 


* мобильную верстку профиля нужно подправить (320px). Скрин https://disk.yandex.ru/i/z5D2G5nctZVX8Q . Там горизонтальный скролл
* в админке должна быть возможность добавить студента. Скрни из фигмы https://disk.yandex.ru/i/6FnhniKQ3I2KEg  . И редактировать кагорты, удалять студентов и тд
* под админом тоже должны быть видны фото студентов https://disk.yandex.ru/i/9bdXrUFavsZEAA . И комментарии к фото 
* Для любых запросов на сервер в **самом конце цепочки** запроса нужно осуществлять проверку на возможные ошибки с помощью `catch`.
* Все события обработчиков нужно **правильно** типизировать. Это легко сделать. Для `onChange` есть тип `ChangeEvent<HTMLInputElement>`, а для `onSubmit` есть `FormEvent<HTMLFormElement>` в библиотеке `react`. Для обычных функций есть тип `SyntheticEvent` для их событий. Тип `Response` для ответа от сервера, тип `KeyboardEvent` для событий клавиатуры, тип `RouteProps` для рутов
* Есть небольшая хитрость: можно всегда проверить, какой тип у функции или события. Для этого нужно навести мышкой на атрибут `onChange`, `onSubmit`, `onClick` и тд. Там высветится типизация того, что принимают эти атрибуты. 
* Нужно правильно удалять обработчики, если это делаете через `removeEventListener`
* Посмотрите еще раз в фигме, что нужно отображать на странице админа. Может, что-то еще добавите в верстку



Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

trashmarket committed:
fixed css form and aded xlsx component

Dumisel committed:
AdminPage структура, комментарии + плашка для пользователя

happy-land committed:
Merge branch 'dev-backup' into nikita-fixed-form

happy-land committed:
Merge pull request #44 from happy-land/nikita-fixed-form

fixed css form and aded xlsx component

happy-land committed:
Merge branch 'dev-backup' into lisa-admin-comments

happy-land committed:
Merge pull request #45 from happy-land/lisa-admin-comments

AdminPage структура, комментарии + плашка для пользователя

happy-land committed:
routes fixed

happy-land committed:
Merge pull request #46 from happy-land/ruslan-routes-fix

исправил ошибку в app-routes.tsx

annasilina committed:
deleted unused file api.ts
deleted unused files and directory utils
added all missing catches for api requests
added private _request method as wrapper above fetch in Api
deleted unused login-form
fixed button at button id (added Button UI component, fixed CSS for button text)
fixed CSS for mobile screen at login page

happy-land committed:
Merge pull request #47 from happy-land/anna-api-fix

deleted unused file api.ts

trashmarket committed:
scroll and file fiter aded

trashmarket committed:
Merge pull request #48 from happy-land/select-add-scroll

Добавил не много по скролу и не много еще с формами

Dumisel committed:
AdminPage - StudentsList

Отрисовала заново лист студентов, его можно редактировать и сортировать.
Нужно добавить фильтр на студентов и комментарии, функция добавления студентов. Исправлены замечания после ревью с комментами.

happy-land committed:
Merge pull request #49 from happy-land/lisa-admin-students

AdminPage - StudentsList

happy-land committed:
admin-comments filtering added

happy-land committed:
admin filters done

happy-land committed:
Merge pull request #50 from happy-land/ruslan-filters

Ruslan filters

happy-land committed:
stopPropagation bug fixed

happy-land committed:
Merge pull request #51 from happy-land/ruslan-reactions-fix

stopPropagation bug fixed

happy-land committed:
file upload draft

happy-land committed:
Merge pull request #52 from happy-land/ruslan-xlsx-test

file upload draft

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
